draft-ietf-tls-dtls-connection-id.md

```diff
@@ -162,158 +161,64 @@ connection identifiers to other parties.
 In DTLS 1.2, connection ids are exchanged at the beginning of the DTLS
 session only. There is no dedicated "connection id update" message
 that allows new connection ids to be established mid-session, because
-DTLS 1.2 in general does not allow post-handshake messages that do not
-themselves begin other handshakes. In DTLS 1.3, which does allow such
-messages, we use post-handshake message to update the connection ID
-{{post-handshake-messages}} and to request new IDs.
+DTLS 1.2 in general does not allow TLS 1.3-style post-handshake messages
+that do not themselves begin other handshakes. DTLS 1.2 peers switch to
+the new record layer format when encryption is enabled.
 
-DTLS 1.2 peers switch to the new record layer format when encryption
-is enabled. The same is true for DTLS 1.3 but since the DTLS 1.3
-enables encryption early in the handshake phase the connection ID will
-be enabled earlier. For this reason, the connection ID needs to go in
-the DTLS 1.3 ServerHello.
-
-
-# Post-Handshake Messages
-
-In DTLS 1.3, if the client and server have negotiated the "connection_id" extension,
-either side can send a new connection ID which it wishes the other side to use
-in a NewConnectionId message:
+# Record Layer Extensions
 
-~~~
-   enum {
-       cid_immediate(0), cid_spare(1), (255)
-   } ConnectionIdUsage;
+This extension is applicable for use with DTLS 1.2 and {{dtls-record12}}
+illustrates the record format.
 
+~~~~
    struct {
-       opaque cid<0..2^8-1>;
-       ConnectionIdUsage usage;
-   } NewConnectionId;
-~~~
-
-cid
-: Indicates the CID which the sender wishes the peer to use.
-
-usage
-: Indicates whether the new CID should be used immediately or is a spare.
-If usage is set to "cid_immediate", then the new CID MUST be used immediately
-for all future records. If it is set to "cid_spare", then either CID MAY
-be used, as described in {{sec-cons}}.
-{:br}
+        ContentType type;
+        ProtocolVersion version;
+        uint16 epoch;
+        uint48 sequence_number;
+        opaque cid[cid_length];               // New field
```

Comment:
I don't think we want to change DTLSPlaintext. This should only be for encrypted records.

Reply:
I might be horribly wrong, but on a re-handshake are we not dealing with DTLSPlaintext's?

Reply:
I don't follow. We will only send CIDs on encrypted data, and therefore it doesn't need to be in this structure.

Reply:
Yes, I get that.  The thing that confuses me then is how to interpret this bit of [RFC5246](https://tools.ietf.org/html/rfc5246#section-7.4) regarding (re-)handshake records:
```
   [...] Handshake messages are supplied to
   the TLS record layer, where they are encapsulated within one or more
   TLSPlaintext structures, which are processed and transmitted as
   specified by the current active session state.
```
And this other bit of [RFC5246](https://tools.ietf.org/html/rfc5246#section-6.2.3):
```
      The encryption and MAC functions translate a TLSCompressed
      structure into a TLSCiphertext.
```
Which is the reason why I originally added CID to both `DTLSPlaintext` and `DTLSCompressed`.

Reply:
Sure, and the way to think about this is that you are adding the CID as part of that transformation.

Reply:
Got it, thanks.

(We still need to say that CID is fed into the MAC algorithm.)

Reply:
> (We still need to say that CID is fed into the MAC algorithm.)

Done in 0b40194

Reply:
Did we conclude that the CID only needs to be added to the DTLSCiphertext (and not to the DTLSPlaintext and the DTLSCompressed)?

Reply:
I got confused by this comment from the 101 minutes:
```
Subodh: Re-read draft and the TLS conn-ID isn't under the authentication?

ekr: Need to write it down in 1.2.
```
which I thought it meant we forgot to put CID in under authentication.

If this is not the case, I can remove the modified MAC computation together with the DTLSCompressed stuff, which was just needed to reference DTLSCompressed.cid in the algorithm.